hack/verify-crd-gen.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCHED_PLUGIN_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${SCHED_PLUGIN_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+cd "${SCHED_PLUGIN_ROOT}"
+
+CRD_OPTIONS="crd:trivialVersions=true,preserveUnknownFields=false"
+
+# Download controller-gen locally
+CONTROLLER_GEN="$(shell pwd)/bin/controller-gen"
+$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0)
+
+# Create a temp directory
+_tmpdir="$(mktemp -d "${SCHED_PLUGIN_ROOT}/_tmp")"
+_crdtmp="${_tmpdir}"
+# Generate CRD
+pushd "${_crdtmp}" > /dev/null 2>&1
+$(CONTROLLER_GEN) ${CRD_OPTIONS} path="./..." output:crd:artifacts:config=${_crdtmp}/crd/bases
+
+popd > /dev/null 2>&1
+
+pushd "${SCHED_PLUGIN_ROOT}" > /dev/null 2>&1
+if ! _out="$(diff -Naupr manifests/coscheduling/ "${_crdtmp}/crd/bases/}")"; then
+    echo "Generated output differs:" >&2
+    echo "${_out}" >&2
+    echo "Verification failed."
```

Comment:
Prefer to explicitly add `exit 1` below; otherwise, this script (if I understand correctly) will always return 0 as you catch the potential error in L30.